app/models/supermarket.rb

```diff
@@ -1,3 +1,13 @@
 class Supermarket < ApplicationRecord
   has_many :items
+
```

Comment:
There are a couple things here that you could do, one of them is to use AR `.joins` on `items: :customer_items` or something of that nature. 

You can also add additional associations like
```ruby
has_many :customers, through: :items
```
This will allow you to start right at `customers` and then you can just call `.distinct`.